lib/blocs/home_page_cubit.dart

```diff
@@ -93,181 +87,67 @@ class HomePageCubit extends Cubit<MapRouteState> {
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(
-            context,
-            "Offline mode is not implemented yet.",
-            false,
-            localization.commonError);
-      } on FetchOnlineRequestException catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(context, localization.commonNoInternet,
-            true, localization.commonError);
-      } on FetchOnlineResponseException catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showOnAndOfflineErrorAlert(context, localization.searchFailLoadingPlan,
-            true, localization.commonError);
-      } catch (e) {
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch plan: $e");
-        _showErrorAlert(context, e.toString());
-      }
-
-      try {
-        currentFetchAdOperation =
-            requestManagerCubit.fetchAd(context, state.toPlace);
-
-        final Ad ad = await currentFetchAdOperation.valueOrCancellation(null);
-        await updateMapRouteState(
-          state.copyWith(ad: ad),
-        );
-      } catch (e, stacktrace) {
-        await updateMapRouteState(
-          MapRouteState(
-              fromPlace: state.fromPlace,
-              toPlace: state.toPlace,
-              isFetching: state.isFetching,
-              showSuccessAnimation: state.showSuccessAnimation,
-              plan: state.plan),
-        );
-        // TODO: Replace by proper error handling
-        // ignore: avoid_print
-        print("Failed to fetch ad: $e");
-        // ignore: avoid_print
-        print(stacktrace);
+      } else {
+        setPlan(plan);
       }
     }
   }
 
-  void _showOnAndOfflineErrorAlert(BuildContext context, String message,
-      bool online, String commonErrorMessage) {
-    showDialog(
-      context: context,
-      builder: (context) {
-        return buildOnAndOfflineErrorAlert(
-          context: context,
-          online: online,
-          title: Text(commonErrorMessage),
-          content: Text(message),
-        );
-      },
-    );
-  }
+// TODO: investigate how works this functions and know why we need it
+  Future<void> fetchAd(String correlationId) async {
+    try {
+      currentFetchAdOperation = requestManager.fetchAd(
+        state.toPlace,
+        correlationId,
+      );
 
-  void _showErrorAlert(BuildContext context, String error) {
-    showDialog(
-      context: context,
-      builder: (context) {
-        return buildErrorAlert(context: context, error: error);
-      },
-    );
+      final Ad ad = await currentFetchAdOperation.valueOrCancellation(null);
+      await updateMapRouteState(
+        state.copyWith(ad: ad),
+      );
+    } catch (e, stacktrace) {
```

Comment:
Can we specify the catched Error?

Reply:
@md-weber probably it will be removed, I never seen before this feature (`fetchAd` function), and I am not sure if this feature will be supported in the future